modules/pubmaticBidAdapter.js

```diff
@@ -217,6 +289,218 @@ function _checkParamDataType(key, value, datatype) {
+              h: params[key].h || params[key].height || (params[key].sizes ? params[key].sizes[1] : undefined)
+            }
+          };
+          break;
+        case 'rating':
+          assetObj = {
+            id: NATIVE_ASSET_ID.RATING,
+            required: params[key].required ? 1 : 0,
+            data: {
+              type: NATIVE_ASSET_DATA_TYPE.rating,
+            }
+          };
+          break;
+        case 'likes':
+          assetObj = {
+            id: NATIVE_ASSET_ID.LIKES,
+            required: params[key].required ? 1 : 0,
+            data: {
+              type: NATIVE_ASSET_DATA_TYPE.likes,
+            }
+          };
+          break;
+        case 'downloads':
+          assetObj = {
+            id: NATIVE_ASSET_ID.DOWNLOADS,
+            required: params[key].required ? 1 : 0,
+            data: {
+              type: NATIVE_ASSET_DATA_TYPE.downloads,
+            }
+          };
+          break;
+        case 'price':
+          assetObj = {
+            id: NATIVE_ASSET_ID.PRICE,
+            required: params[key].required ? 1 : 0,
+            data: {
+              type: NATIVE_ASSET_DATA_TYPE.price,
+            }
+          };
+          break;
+        case 'saleprice':
+          assetObj = {
+            id: NATIVE_ASSET_ID.SALEPRICE,
+            required: params[key].required ? 1 : 0,
+            data: {
+              type: NATIVE_ASSET_DATA_TYPE.saleprice,
+            }
+          };
+          break;
+        case 'phone':
+          assetObj = {
+            id: NATIVE_ASSET_ID.PHONE,
+            required: params[key].required ? 1 : 0,
+            data: {
+              type: NATIVE_ASSET_DATA_TYPE.phone,
+            }
+          };
+          break;
+        case 'address':
+          assetObj = {
+            id: NATIVE_ASSET_ID.DATA,
+            required: params[key].required ? 1 : 0,
+            data: {
+              type: NATIVE_ASSET_DATA_TYPE.address,
+            }
+          };
+          break;
+        case 'desc2':
+          assetObj = {
+            id: NATIVE_ASSET_ID.DESC2,
+            required: params[key].required ? 1 : 0,
+            data: {
+              type: NATIVE_ASSET_DATA_TYPE.desc2,
+            }
+          };
+          break;
+        case 'displayurl':
+          assetObj = {
+            id: NATIVE_ASSET_ID.DISPLAYURL,
+            required: params[key].required ? 1 : 0,
+            data: {
+              type: NATIVE_ASSET_DATA_TYPE.displayurl,
+            }
+          };
+          break;
+        case 'ctatext':
+          assetObj = {
+            id: NATIVE_ASSET_ID.CTATEXT,
+            required: params[key].required ? 1 : 0,
+            data: {
+              type: NATIVE_ASSET_DATA_TYPE.ctatext,
+            }
+          };
+          break;
+      }
+    }
+    nativeRequestObject.assets[nativeRequestObject.assets.length] = assetObj;
+  };
+  NATIVE_MINIMUM_REQUIRED_IMAGE_ASSETS.forEach(ele => {
```

Comment:
add some explanation in comment about what we are doing here

Reply:
Done